### PR TITLE
test(select): add e2e tests to Select

### PR DESCRIPTION
**Pull request by blackswanny**
add e2e tests to Select

### CODE REVIEW

File: src/select/e2e.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  selectDropDown: '[role="listbox"]',
+  dropDownOption: '[role="option"]',
+  selectedList: '[role="list"]',
+  selectedOption: '[role="listitem"]',
+  searchType: '[aria-autocomplete="list"]',
+  selectType: '[aria-autocomplete="none"]',
+  expandedDropDown: '[aria-expanded="true"]',
+};
+
+describe('The select component', () => {
```
Comment (gergelyke) [inline src/select/e2e.js]: can you add the accessibility checks?

File: src/select/select.js [hunk 2]
```diff
@@ -457,6 +467,16 @@ class Select extends React.Component<PropsT, StatelessStateT> {
       ),
     };
   }
+
+  getAccessibilityProps() {
```
Comment (gergelyke) [inline src/select/select.js]: shouldn't we address these in a separate PR?

File: src/select/select.js [hunk 3]
```diff
@@ -296,7 +303,7 @@ class Select extends React.Component<PropsT, StatelessStateT> {
     const {type, disabled, selectedOptions} = this.props;
     const multiple = this.isMultiple();
     return (
-      <React.Fragment>
+      <div role="list">
```
Comment (nadiia) [inline src/select/select.js]: With adding an additional element we need to add it to the overrides API to be replaceable

File: src/select/select.js (hunk 2 above)
Comment (blackswanny) [reply src/select/select.js]: https://github.com/uber-web/baseui/pull/409

File: src/select/select.js (hunk 3 above)
Comment (blackswanny) [reply src/select/select.js]: added here #409

File: src/select/e2e.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    browser.end(() => done());
+  });
+
+  it('opens dropdown menu when click on select input', browser => {
+    goToUrl({
+      suite,
+      test: scenarios.SELECT,
+      browser,
+    })
+      .waitForElementVisible('body', 1000)
```
Comment (nadiia) [inline src/select/e2e.js]: We should wait on the `selectors.selectInput` to be visible instead

File: src/select/e2e.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      .click(selectors.selectInput)
+      .waitForElementPresent(selectors.selectDropDown, 1000)
+      .click(
+        selectors.selectDropDown +
+          ' ' +
+          selectors.dropDownOption +
+          ':first-child',
```
Comment (nadiia) [inline src/select/e2e.js]: string template instead of string concatenation

File: src/select/e2e.js (hunk 4 above)
Comment (blackswanny) [reply src/select/e2e.js]: ok

File: src/select/e2e.js (hunk 5 above)
Comment (blackswanny) [reply src/select/e2e.js]: ok

**Comment by blackswanny** [comment]
> Left a few comments but other than that looks great!
> Could you add a few tests for the multiselect?

@nadiia added

File: src/select/e2e.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    goToUrl({
+      suite,
+      test: scenarios.MULTI_SELECT_SEARCH,
+      browser,
+    })
+      .waitForElementVisible(selectors.selectInput, 1000)
+      .setValue(selectors.selectInput, 'dark', () => {
+        browser
+          .waitForElementPresent(selectors.selectDropDown, 100)
+          .pause(1000)
```
Comment (nadiia) [inline src/select/e2e.js]: Can we increase the wait time to `1000` on the line 139 instead of `pause` here?

**Comment by nadiia** [comment]
Oh, the tests on CI fail, please make sure it passes before merging.

File: src/select/e2e.js (hunk 6 above)
Comment (blackswanny) [reply src/select/e2e.js]: that will not work as `pause`, which works to wait for async call, which we emulate here. Cause DropDown is shown immediately with loading spinner, but no data yet there

Comment (nadiia) [reply src/select/e2e.js]: Oh, I see. Makes sense. Another approach would be to wait until a specific option is visible. If we keep `pause` versus `wait for an option`, could you add a comment to it about the async call and a loading state here?

**Pull request merged**